lib/ansible/modules/commands/psexec.py

```diff
@@ -0,0 +1,527 @@
+    limited = module.params['limited']
+    use_system = module.params['use_system']
+    interactive = module.params['interactive']
+    interactive_session = module.params['interactive_session']
+
+    priority = {
+        "above_normal": ProcessPriority.ABOVE_NORMAL_PRIORITY_CLASS,
+        "below_normal": ProcessPriority.BELOW_NORMAL_PRIORITY_CLASS,
+        "high": ProcessPriority.HIGH_PRIORITY_CLASS,
+        "idle": ProcessPriority.IDLE_PRIORITY_CLASS,
+        "normal": ProcessPriority.NORMAL_PRIORITY_CLASS,
+        "realtime": ProcessPriority.REALTIME_PRIORITY_CLASS
+    }[module.params['priority']]
+    show_ui_on_win_logon = module.params['show_ui_on_win_logon']
+
+    process_timeout = module.params['process_timeout']
+    stdin = module.params['stdin']
+
+    if connection_username is None or connection_password is None and \
+            not HAS_KERBEROS:
+        module.fail_json(msg='The gssapi python module with the GGF extension '
+                             'is required for Kerberos authentication')
+
+    win_client = client.Client(server=hostname, username=connection_username,
+                               password=connection_password, port=port,
+                               encrypt=encrypt)
+
+    try:
+        win_client.connect(timeout=connection_timeout)
+    except SMBAuthenticationError as exc:
+        module.fail_json(msg='Failed to authenticate over SMB: %s'
+                             % to_native(exc))
+    except SMBResponseException as exc:
+        module.fail_json(msg='Received unexpected SMB response when opening '
+                             'the connection: %s' % to_native(exc))
+    except SMBException as exc:
+        module.fail_json(msg=to_native(exc), exception=exc)
+    except PDUException as exc:
+        module.fail_json(msg='Received an exception with RPC PDU message: %s'
+                             % to_native(exc))
+    except SCMRException as exc:
+        module.fail_json(msg='Received an exception when dealing with SCMR on '
+                             'the Windows host: %s' % to_native(exc))
+    except PypsexecException as exc:
+        module.fail_json(msg=to_native(exc), exception=exc)
+
+    # create PAExec service and run the process
+    result['changed'] = True
+    b_stdin = to_bytes(stdin, encoding='utf-8') if stdin else None
+    run_args = dict(
+        executable=executable, arguments=arguments, asynchronous=asynchronous,
+        load_profile=load_profile, interactive_session=interactive_session,
+        run_elevated=elevated, run_limited=limited,
+        username=process_username, password=process_password,
+        use_system_account=use_system, working_dir=working_directory,
+        priority=priority, show_ui_on_win_logon=show_ui_on_win_logon,
+        timeout_seconds=process_timeout, stdin=b_stdin
+    )
+    if not module.check_mode:
+        try:
+            win_client.create_service()
+        except PDUException as exc:
```

Comment:
Since the error handling is effectively the same beyond slight tweaks to the messaging, you could collapse all this with 
```python
except (SCMRException,PDUException,PypsexecException,SMBException) as exc:
...
```